### DISCUSSION TITLE
More detailed stepper configuration

### DISCUSSION THREAD

**Discussion by JulianGroeli** [Guilouz/Klipper-Flsun-Speeder-Pad#69] [Ideas]
It would be nice to add a bit more detail about stepper configuration, and maybe even change the defaults.

I was having real trouble with accuracy, even after doing the advanced delta configuration. Basically all of the 90 deg corners in the XY plane were being rounded, like the printer could not make a sharp turn - both on internal and external corners. This was mainly apparent when printing rectangular parts. Organic and rounded shapes looked OK. 

After lots of experimenting and reading more of the klipper documentation, I have found much better results by setting the "spreadCycle" mode in each stepper config section of printer.cfg, instead of the default "stealthChop" mode. 
`stealthchop_threshold: 0`

stealthChop is a bit quieter than spreadCycle, but it's well worth it for the increased accuracy, especially on a delta machine. 

Reading more about the TMC driver config (here: https://www.klipper3d.org/TMC_Drivers.html) I also see that hold current is not recommended and TMC interpolate is also not recommended for klipper. I'm still testing these settings to see their effects.

**Comment by 808mp5** [comment]
The rounded corners are because the included printer.cfg has "pressure_advance" commented out.. you have to calibrate the pressure advance or just un-comment it and you will see an improvement.
